### PR TITLE
binder: plan not optimal for count(*) with filter

### LINKED ISSUE #482: binder: plan not optimal for count(*) with filter
```
PhysicalProjection: exprs [InputRef #0]
  PhysicalSimpleAgg: 1 agg calls
    PhysicalTableScan: table #9, columns [0], with_row_handler: true, is_sorted: false, expr: Gt(InputRef #0, Int32(5) (const))
```

The plan is not optimal. As we already have column 0 scanned, we don't need row handler.

We can refine the binder logic later to handle this case, and get this PR merged at first. Thanks!

_Originally posted by @skyzh in https://github.com/risinglightdb/risinglight/pull/447#discussion_r807561493_

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xiaoyong-z <a997647204@gmail.com>

close #482 

For select statement, set with_row_handler to be true when "with_row_handler = true" and "column_ids.is_empty()".
For delete statement, always set with_row_handler to be true.

### CODE REVIEW

**Review by xxchan — APPROVED** [review]
Can you show some example plans?

File: src/logical_planner/select.rs [hunk 1]
```diff
@@ -140,7 +141,7 @@ impl LogicalPlaner {
                         *ref_id,
                         column_ids.to_vec(),
                         column_descs.to_vec(),
-                        with_row_handler,
+                        is_delete || (with_row_handler && column_ids.is_empty()),
```
[the remark this answers is not in the document]
Comment (xiaoyong-z) [reply src/logical_planner/select.rs:144]: Well, the "is_delete" flag indicates whether the plan is from a delete statement.
For "delete" statement, always set with_row_handler to be true as "is delete" flag is true for "delete" statement.
For "select" statement, the "is_delete" flag is false, and then we need to further check if the with_row_handler is true and column_ids is empty. Here i use the && insteand of || is to ensure that when we already have some column scans, we don't need the `row_handler`, the picture following points out this problem.

image

As you metioned, I'm not sure if we need to add more logic in optimizer part.

> And is the `with_row_handler` parameter still needed? Can we derive it directly here?
> 
> BTW, I think it is not enough to only handle it here in planner. Maybe in optimizer instead, as mentioned in #503.

[the remark this answers is not in the document]
Comment (xiaoyong-z) [reply src/logical_planner/select.rs:144]: I think for "delete" statement, it is not needed. For "select" statement, may be it is still needed? 
> And is the `with_row_handler` parameter still needed? Can we derive it directly here?

For this bug in the #482, I think handle it in the planner is enough? In #503, i didn't see how to handle this part in optimizer, can you give me more insights?
> BTW, I think it is not enough to only handle it here in planner. Maybe in optimizer instead, as mentioned in #503.

[the remark this answers is not in the document]
Comment (xxchan) [reply src/logical_planner/select.rs:144]: I'm wondering why planner still needs to compute `with_row_handler` when it is a given parameter? Does it mean that the argument is wrong?

[the remark this answers is not in the document]
Comment (xiaoyong-z) [reply src/logical_planner/select.rs]: Now the computation of `with_row_handler` is only in `plan_select` function.

**Pull request merged**